compile time warning when building net_sockets.h on Android

### Description
- Type: Bug
- Priority: Minor
Raised in the [forum](https://forums.mbed.com/t/how-to-avoid-warning-passing-int-to-parameter-of-type-socklen-t-aka-unsigned-int-converts-between-pointers-to-integer-types-with-different-sign-wpointer-sign/6638) by pii

---------------------------------------------------------------
## Bug

**OS**  
Android

**mbed TLS build:**  
Version: 2.16.3
OS version: Android/sdk/ndk-bundle/toolchains/llvm/prebuilt/darwin-x86_64/

**Expected behavior**   
Build finishes without any warning
**Actual behavior**  
Build warning because neither `#if defined(__socklen_t_defined) || defined(_SOCKLEN_T) ||  \
    defined(_SOCKLEN_T_DECLARED) || defined(__DEFINED_socklen_t)` is defined:
~~~
External/.build_area/mbedtls-2.16.3/library/net_sockets.c:332:37: warning: passing 'int *'
      to parameter of type 'socklen_t *' (aka 'unsigned int *') converts between pointers to integer types with
      different sign [-Wpointer-sign]
                    (void *) &type, &type_len ) != 0 ||
                                    ^~~~~~~~~
/Users/pi/Library/Android/sdk/ndk-bundle/toolchains/llvm/prebuilt/darwin-x86_64/sysroot/usr/include/sys/socket.h:311:92: note: 
      passing argument to parameter '__value_length' here
__socketcall int getsockopt(int __fd, int __level, int __option, void* __value, socklen_t* __value_length);
                                                                                           ^
External/.build_area/mbedtls-2.16.3/library/net_sockets.c:342:80: warning: passing 'int *'
      to parameter of type 'socklen_t *' (aka 'unsigned int *') converts between pointers to integer types with
      different sign [-Wpointer-sign]
                                             (struct sockaddr *) &client_addr, &n );
~~~
`getsockopt` receives 'socklen_t *' (aka 'unsigned int *') , however since none of the precompilation checks is defined, `int` is sent as parameter

--- a/ChangeLog.d/android-socklen_t.txt
+++ b/ChangeLog.d/android-socklen_t.txt
@@ -1,0 +1,3 @@
+Bugfix
+   * Use socklen_t on Android and other POSIX-compliant system
+

--- a/library/net_sockets.c
+++ b/library/net_sockets.c
@@ -318,7 +318,7 @@ int mbedtls_net_accept( mbedtls_net_context *bind_ctx,
 
 #if defined(__socklen_t_defined) || defined(_SOCKLEN_T) ||  \
     defined(_SOCKLEN_T_DECLARED) || defined(__DEFINED_socklen_t) || \
-    defined(socklen_t)
+    defined(socklen_t) || (defined(_POSIX_VERSION) && _POSIX_VERSION >= 200112L)
     socklen_t n = (socklen_t) sizeof( client_addr );
     socklen_t type_len = (socklen_t) sizeof( type );
 #else

